sql/testdata/timedate

```diff
@@ -0,0 +1,38 @@
+statement ok
+CREATE TABLE t (
+  a TIMESTAMP PRIMARY KEY,
+  b DATE,
+  c INTERVAL,
+  CONSTRAINT bidx UNIQUE (b),
+  CONSTRAINT cidx UNIQUE (c)
+)
+
+# TODO(vivek) replace type casting in this file.
+statement ok
+INSERT INTO t VALUES
+  ('2015-08-30 03:34:45.34567'::timestamp, '2015-08-30'::date, '34h2s'::interval),
+  ('2015-08-25 04:45:45.53453'::timestamp, '2015-08-25'::date, '2h45m2s234ms'::interval),
+  ('2015-08-29 23:10:09.98763'::timestamp, '2015-08-29'::date, '234h45m2s234ms'::interval)
+
+query TTT
+SELECT * FROM t WHERE a = '2015-08-25 05:45:45.53453+01:00'::timestamp
```

Comment:
I was planning on adding implicit casting to this change, but then it kept getting bigger. I'm going to follow up this change with two more PRs. 1. support for implicit casting from string to the different types. 2. important functions like CURRENT_{DATE, TIMESTAMP}